drivers/sensor/mc3419/mc3419_trigger.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2023 Linumiz
+ */
+
+#include "mc3419.h"
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(MC3419, CONFIG_SENSOR_LOG_LEVEL);
+
+static void mc3419_gpio_callback(const struct device *dev,
+				 struct gpio_callback *cb,
+				 uint32_t pin_mask)
+{
+	struct mc3419_driver_data *data = CONTAINER_OF(cb,
+					  struct mc3419_driver_data, gpio_cb);
+
+	const struct mc3419_config *cfg = data->dev->config;
+
+	if ((pin_mask & BIT(cfg->int_gpio.pin)) == 0U) {
+		return;
+	}
+
+#if defined(CONFIG_MC3419_TRIGGER_OWN_THREAD)
+	k_sem_give(&data->trig_sem);
+#else
+	k_work_submit(&data->work);
+#endif
+}
+
+static void mc3419_process_int(const struct device *dev)
+{
+	int ret = 0;
+	const struct mc3419_config *cfg = dev->config;
+	const struct mc3419_driver_data *data = dev->data;
+	uint8_t int_source = 0;
+
+	ret = i2c_reg_read_byte_dt(&cfg->i2c, MC3419_REG_INT_STATUS, &int_source);
+	if (ret < 0) {
+		goto exit;
+	}
+
+	if (int_source & MC3419_DATA_READY_MASK) {
+		if (data->handler[MC3419_TRIG_DATA_READY]) {
+			data->handler[MC3419_TRIG_DATA_READY](dev,
+				data->trigger[MC3419_TRIG_DATA_READY]);
+		}
+	}
+
+	if (int_source & MC3419_ANY_MOTION_MASK) {
```

Comment:
`else if`  or both interrupt can occur at the same time?

Reply:
decision is based on status register and will be cleared on read so good to check both imo